packages/go/analysis/ad/adcs.go

```diff
@@ -161,6 +165,160 @@ func PostADCSESC3(ctx context.Context, tx graph.Transaction, outC chan<- analysi
+			return true
+		}
+		//Since CertTemplateControllers only contains first degree controllers, if the principal is a group we do a group expansion to check for certTemplate control through nested members
+		if controller.Kinds.ContainsOneOf(ad.Group) {
+			expanded := groupExpansions.Cardinality(controller.ID.Uint32()).(cardinality.Duplex[uint32])
+			if expanded.Contains(principalID) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func checkEmailValidity(node *graph.Node, validCertTemplates []*graph.Node, groupExpansions impact.PathAggregator, cache ADCSCache) cardinality.Duplex[uint32] {
+	var (
+		results   = cardinality.NewBitmap32()
+		email, _  = node.Properties.Get(common.Email.String()).String()
+		exception = false
+	)
+	if email == "" {
+		for _, certTemplate := range validCertTemplates {
+			if principalControlsCertTemplate(node.ID.Uint32(), certTemplate, groupExpansions, cache) {
+				var (
+					schemaVersion, _              = certTemplate.Properties.Get(ad.SchemaVersion.String()).Float64()
+					subjectAltRequireEmail, _     = certTemplate.Properties.Get(ad.SubjectAltRequireEmail.String()).Bool()
+					subjectRequireEmail, _        = certTemplate.Properties.Get(ad.SubjectRequireEmail.String()).Bool()
+					subjectAltRequireDNS, _       = certTemplate.Properties.Get(ad.SubjectAltRequireDNS.String()).Bool()
+					subjectAltRequireDomainDNS, _ = certTemplate.Properties.Get(ad.SubjectAltRequireDomainDNS.String()).Bool()
+				)
+				if node.Kinds.ContainsOneOf(ad.User) && (subjectAltRequireDNS || subjectAltRequireDomainDNS) {
+					continue
+				}
+				if (!subjectAltRequireEmail && !subjectRequireEmail) || schemaVersion == 1 {
+					exception = true
+				}
+			}
+		}
+		if exception {
+			//Principal does not have an email set but at least one of the certTemplates it controls does not require it so add it to the list of principals enabled for ESC6a
+			results.Add(node.ID.Uint32())
+		}
+	} else {
+		//Principal has an email set so add it to the list of principals enabled for ESC6a
+		results.Add(node.ID.Uint32())
+	}
+	return results
+}
+
+func filterTempResultsForESC6a(tx graph.Transaction, tempResults cardinality.Duplex[uint32], groupExpansions impact.PathAggregator, validCertTemplates []*graph.Node, cache ADCSCache) cardinality.Duplex[uint32] {
+	principalsEnabledForESC6a := cardinality.NewBitmap32()
+
+	tempResults.Each(func(value uint32) (bool, error) {
+		sourceID := graph.ID(value)
+
+		if resultNode, err := tx.Nodes().Filter(query.Equals(query.NodeID(), sourceID)).First(); err != nil {
+			return true, nil
+		} else {
+			if resultNode.Kinds.ContainsOneOf(ad.Group) {
+				//A Group will be added to the list since it requires no further conditions
+				principalsEnabledForESC6a.Add(value)
+
+				//Obtain the members of this group so that we can handle principals with control over valid certTemplates through nested membership
+				expanded := groupExpansions.Cardinality(value).(cardinality.Duplex[uint32])
```

Comment:
If this is a group, should we even expand its membership? All members of this group now have the permission via pathing through the group, so we're explicitly destroying shortcutting by doing this

Reply:
@JonasBK @StephenHinck what are your thoughts on this?

Reply:
Right.. It was actually me who told Uli that for users/computers without the email property, we should check if they have enrollment rights on any of the identified ESC6 CertTemplates that don’t require email. I told him we should make sure to check through group membership as well – and that was a mistake. We should only check if the user/computer has enrollment rights directly. 
So that means User7 should not get an edge in this harness:
![image](https://github.com/SpecterOps/BloodHound/assets/12843299/361ef6cf-158d-4a98-9d96-1679a2be8bae)

So sorry for misguiding you on this one @urangel!

Reply:
I've updated this logic to do a cross product with each certTemplate controller against the collection of enterpriseCA enrollers as discussed earlier. This reflects the edge creation that we want to happen without having to unroll group membership.